src/components/Conversation/EmptyMessageFeed.tsx

```diff
@@ -87,6 +95,40 @@ export default function EmptyMessageFeed({ onAddMemberClick }: Props) {
     )
 }
 
+function TokenGatedMessage() {
+    const { tokenAddress, tokenIds, tokenType, stakingEnabled } = useSelectedRoom() || {}
+
+    if (!tokenType || tokenIds) {
+        return null
+    }
+    const tokenMetadata = useTokenMetadata(tokenAddress, tokenIds || [])
```

Comment:
Hooks have to go before any `return` above them, both (`useTokenMetadata` & `useDisplayUsername`).

React will complain about hooks being used in different order otherwise (it's a runtime error so a bit tricky to catch at first).

In other words,
```tsx
const tokenMetadata = useTokenMetadata(tokenAddress, tokenIds || [])

// …

const displayTokenAddress = useDisplayUsername(tokenAddress, {
    fallback: tokenAddress,
})

// The following was on top before.
if (!tokenType || tokenIds) {
    return null
}
```